features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -211,21 +231,28 @@ DRESULT disk_ioctl(BYTE pdrv, BYTE cmd, void *buff)
             if (_ffs[pdrv] == NULL) {
                 return RES_NOTRDY;
             } else {
-                DWORD count = _ffs[pdrv]->size() / _ffs[pdrv]->get_erase_size();
-                *((DWORD*)buff) = count;
+                *((DWORD*)buff) = disk_get_sector_count(pdrv);
                 return RES_OK;
             }
         case GET_SECTOR_SIZE:
             if (_ffs[pdrv] == NULL) {
                 return RES_NOTRDY;
             } else {
-                WORD size = _ffs[pdrv]->get_erase_size();
-                *((WORD*)buff) = size;
+                *((WORD*)buff) = disk_get_sector_size(pdrv);
                 return RES_OK;
             }
         case GET_BLOCK_SIZE:
             *((DWORD*)buff) = 1; // default when not known
             return RES_OK;
+        case CTRL_TRIM:
+            if (_ffs[pdrv] == NULL) {
+                return RES_NOTRDY;
+            } else {
+                DWORD *sectors = (DWORD*)buff;
+                DWORD ssize = disk_get_sector_size(pdrv);
+                int err = _ffs[pdrv]->trim(sectors[0]*ssize, (sectors[1]-sectors[0]+1)*ssize);
```

Comment:
Arguments to erase function were start sector, and total size to erase. Please clarify what are the arguments of trim? sectors[0] / sectors[1] what do they contain?

Reply:
Ah sure: http://elm-chan.org/fsw/ff/doc/dioctl.html

> The sector block is specified by a DWORD array `{<start sector>, <end sector>}`

So sectors[0] is the start sector, and sectors[1] is the end sector (inclusive) of the trim operation.